src/test/java/org/jenkinsci/plugins/oic/OicSecurityRealmFipsTest.java

```diff
@@ -0,0 +1,98 @@
+package org.jenkinsci.plugins.oic;
+
+import hudson.util.FormValidation;
+import hudson.util.Secret;
+import java.io.IOException;
+import jenkins.security.FIPS140;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.reactor.ReactorException;
+import org.jvnet.hudson.test.FlagRule;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.WithoutJenkins;
+import org.jvnet.hudson.test.recipes.LocalData;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThrows;
+
+public class OicSecurityRealmFipsTest {
+
+    @ClassRule
+    public static FlagRule<String> fipsFlag = FlagRule.systemProperty(FIPS140.class.getName() + ".COMPLIANCE", "true");
+
+    @Rule
+    public NonFailingOnStartupJenkinsRule j = new NonFailingOnStartupJenkinsRule();
+
+    @Test
+    @WithoutJenkins
+    public void settingNonCompliantValuesNotAllowedTest() throws IOException {
+        OicSecurityRealm realm = new OicSecurityRealm("clientId", Secret.fromString("secret"), null, false);
+        IllegalArgumentException ex = assertThrows(
+                IllegalArgumentException.class,
+                () -> new OicSecurityRealm("clientId", Secret.fromString("secret"), null, true));
+        assertThat(
+                "Exception contains the reason",
+                ex.getLocalizedMessage(),
+                containsString("SSL verification can not be disabled"));
+        realm.setDisableTokenVerification(false);
+        ex = assertThrows(IllegalArgumentException.class, () -> realm.setDisableTokenVerification(true));
+        assertThat(
+                "Exception contains the reason",
+                ex.getLocalizedMessage(),
+                containsString("Token verification can not be disabled"));
+    }
+
+    @Test
+    @WithoutJenkins
+    public void validationWarnsOfInvalidValuesTest() {
+        OicSecurityRealm.DescriptorImpl descriptor = new OicSecurityRealm.DescriptorImpl();
+        FormValidation response = descriptor.doCheckDisableSslVerification(true);
+        assertThat("Shows an error", response.kind, is(FormValidation.Kind.ERROR));
+        assertThat(
+                "States SSL verification can not be disabled",
+                response.getMessage(),
+                containsString("SSL verification can not be disabled"));
```

Comment:
NIT: these can be combined, see https://github.com/jenkinsci/oic-auth-plugin/blob/c7c0c06e8af5510f3359c0ec57ed0bbca6bb0529/src/test/java/org/jenkinsci/plugins/oic/OicServerWellKnownConfigurationTest.java#L49-L59